test/src/main.rs

```diff
@@ -148,13 +148,19 @@ fn main() {
     let mut test_results = Vec::new();
     let mut worker_running = worker_count;
     let mut done_specs = 0;
+    let mut started_sequential = false;
     while worker_running > 0 {
         if max_time > 0 && start_time.elapsed().as_secs() > max_time {
             // shutdown, specs running to long
             workers.shutdown();
             break;
         }
 
+        if worker_running == 1 && !started_sequential {
```

Comment:
This seems the last spec job may still have overlapping with the sequential spec job, suppose the `msg` is `Notify::Start`, means last job is starting executing, and we start `workers.start_sequencial()`, one sequential job will start, so there is a overlapping.

Reply:
add a check `&& msg != Notify::Start` ?

Reply:
> This seems the last spec job may still have overlapping with the sequential spec job, suppose the `msg` is `Notify::Start`, means last job is starting executing, and we start `workers.start_sequencial()`, one sequential job will start, so there is a overlapping.

`workers.start_sequencial()` will only be executed once.

Reply:
I know it will be executed once, I mean when `worker_running == 1 && !started_sequential` is true, msg maybe `Notify::Start`, the last spec job is starting to run, not mean the last job has just finished, if we start sequential job at this time, there maybe a overlapping with last job?

Reply:
seems it's ok, `workers.start_sequencial()` only set a flag.

Reply:
When `worker_running == 1`, only one worker (the sequential worker) should be running, and all other parallel workers should have exited.

> `msg` could be `Notify::Start`
If `msg` is `Notify::Start`, the sequential worker might be executing `self.run_spec(spec.as_ref(), 0);` at that moment.
A worker can execute only one spec at a time.
Therefore, there is no chance of overlap.

Reply:
the last running worker must be the `sequencial_worker`...